### PR TITLE
Use docker compose v2 by default in `test-integration-docker-compatibility`

### LINKED ISSUE #1577: test-integration-docker-compatibility: switch away from `docker-compose` (v1) to `docker compose` (v2)
https://github.com/containerd/nerdctl/blob/dcb6a397e6b637d72fa80faca2b61d90f9a795d4/pkg/testutil/testutil.go#L595

### PR DESCRIPTION
**Pull request by djdongjin**
Fix #1577

Signed-off-by: Jin Dong <djdongjin95@gmail.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@djdongjin Do you still plan to work on this?

**Comment by djdongjin** [comment]
> @djdongjin Do you still plan to work on this?

@AkihiroSuda the PR is ready for review, thanks

File: cmd/nerdctl/compose_build_linux_test.go [hunk 1]
```diff
@@ -32,8 +32,6 @@ services:
   svc0:
     build: .
     image: %s
-    ports:
-    - 8080:80
```
Comment (AkihiroSuda) [inline cmd/nerdctl/compose_build_linux_test.go:36]: ?

File: pkg/composer/serviceparser/serviceparser.go [hunk 2]
```diff
@@ -47,6 +48,10 @@ const (
 	ComposeCosignCertificateOidcIssuerRegexp = "x-nerdctl-cosign-certificate-oidc-issuer-regexp"
 )
 
+// Separator is used for naming components (e.g., service image or container)
+// https://github.com/docker/compose/blob/8c39b5b7fd4210a69d07885835f7ff826aaa1cd8/pkg/api/api.go#L483
+var Separator = "-"
```
Comment (AkihiroSuda) [inline pkg/composer/serviceparser/serviceparser.go:53]:
```suggestion
const Separator = "-"
```

Comment (djdongjin) [reply pkg/composer/serviceparser/serviceparser.go]: fixed

File: pkg/testutil/testutil.go [hunk 3]
```diff
@@ -663,9 +663,15 @@ func newBase(t *testing.T, ns string) *Base {
...
-			t.Fatal(err)
+		// use `docker compose` subcommand (usually compose v2) if exist,
+		// otherwise fall back to `docker-compose` (usually compose v1)
+		if err := exec.Command("docker", "compose", "version").Run(); err == nil {
+			base.ComposeBinary = ""
+		} else {
+			base.ComposeBinary, err = exec.LookPath("docker-compose")
+			if err != nil {
+				t.Fatal(err)
+			}
```
[the remark this answers is not in the document]
Comment (djdongjin) [reply pkg/testutil/testutil.go]: removed

File: cmd/nerdctl/compose_run_linux_test.go [hunk 4]
```diff
@@ -51,6 +51,7 @@ services:
 	t.Logf("projectName=%q", projectName)
 	defer base.ComposeCmd("-f", comp.YAMLFullPath(), "down", "-v").Run()
 
+	defer base.Cmd("rm", "-f", "-v", containerName).Run()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/compose_run_linux_test.go:54]: Isn't `defer base.ComposeCmd("-f", comp.YAMLFullPath(), "down", "-v").Run()` enough?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: cmd/nerdctl/compose_run_linux_test.go (hunk 4 above)
Comment (djdongjin) [reply cmd/nerdctl/compose_run_linux_test.go:54]: It didn't work in docker (compose), e.g., containers from `docker compose run` (i.e., not from `docker compose up`) won't be covered by `docker compose down`:

```shell
$ cat docker-compose.yaml
version: '3.1'

services:
  svc0:
    image: nginx
  svc1:
    image: rabbitmq

# docker compose down cannot down container from docker compose run
$ docker compose run -d svc0
[+] Running 1/0
 ✔ Network tmp_default  Created                                                                                                                                                                                                                                                                                                       0.0s
103f148596f2d629f2b875d83da6ac5ea3682ad85b8c61d58e73ddc7931f3b11
$ docker compose down -v
[+] Running 1/0
 ✘ Network tmp_default  Error                                                                                                                                                                                                                                                                                                         0.0s
failed to remove network tmp_default: Error response from daemon: error while removing network: network tmp_default id 3795360bd3e1e9a11def2a76f140597f328009fb101bd7d67490bdeebb8b54f9 has active endpoints
$ docker ps
CONTAINER ID   IMAGE     COMMAND                  CREATED          STATUS          PORTS     NAMES
103f148596f2   nginx     "/docker-entrypoint.…"   24 seconds ago   Up 23 seconds   80/tcp    tmp-svc0-run-db8a05f1f3e6

# works fine if from docker compose up
$ docker compose up -d
[+] Running 2/2
 ✔ Container tmp-svc1-1  Started                                                                                                                                                                                                                                                                                                      0.2s
 ✔ Container tmp-svc0-1  Started                                                                                                                                                                                                                                                                                                      0.3s
$ docker compose down -v
[+] Running 3/2
 ✔ Container tmp-svc0-1  Removed                                                                                                                                                                                                                                                                                                      0.1s
 ✔ Container tmp-svc1-1  Removed                                                                                                                                                                                                                                                                                                      6.3s
 ✔ Network tmp_default   Removed
```

**Pull request merged**